ingenialink/ethercat/servo.py

```diff
@@ -318,9 +331,11 @@ def _on_emcy(self, emergency_msg: "pysoem.Emergency") -> None:
     def set_pdo_map_to_slave(self, rpdo_maps: list[RPDOMap], tpdo_maps: list[TPDOMap]) -> None:
         for rpdo_map in rpdo_maps:
             if rpdo_map not in self._rpdo_maps:
+                rpdo_map.slave = self
```

Comment:
This implementation worries me. While it's currently the only way to check if the drive is operational, it could be problematic. Consider this scenario:

1. Connect an ECAT drive
2. Create some PDOs and save the PDO maps in variables
3. Disconnect the drive
4. Try to create an item in the PDO map

In step 4, the CdefSlave object likely won't exist, which could cause a crash.